python_modules/dagster/dagster/_core/definitions/partition.py

```diff
@@ -217,7 +217,22 @@ def __lt__(self, other: "ScheduleType") -> bool:
         return self.ordinal < other.ordinal
 
 
+@whitelist_for_serdes
+class PartitionsSubsetType(Enum):
```

Comment:
I'm wondering if there's a way to avoid introducing this meta-concept and the associated complexity.

What if we serialized `partitions_def.__class__.__name__` as part of `partitions_def_class_names_by_asset_key` instead of  `subset_types_by_asset_key` and said `can_deserialize` is `False` if `partitions_def_class_names_by_asset_key[asset_key] != asset_graph.get_partitions_def(asset_key).__class__.__name__`?

Reply:
I considered that too, though if the partitions subset class name changed then that would unexpectedly make all historic backfills display with the "partitions def has changed" error in the UI.

I figured that if we did this instead, if class names changed we would still have a backcompat route

Reply:
I was thinking the PartitionsDefinition subclass - if those class names change, we'd have much bigger problems on our hands